### PR TITLE
T14783 integration suite rework

### PR DESCRIPTION
**Pull request by niden**
Hello!

* Type: enhancement - tests
* Link to issue: #14779 

**In raising this pull request, I confirm the following:**

- [x] I have read and understood the [Contributing Guidelines](https://github.com/phalcon/cphalcon/blob/master/CONTRIBUTING.md)
- [x] I have checked that another pull request for this purpose does not exist
- [x] I wrote some tests for this PR
- [ ] I have updated the relevant CHANGELOG
- [ ] I have created a PR for the [documentation](https://github.com/phalcon/docs) about this change

Created `database` suite to be run with `--env mysql` parameter against the target RDBMS
Kept integration tests where they are unless they access the database
Moved potential db tests in a `tocheck` folder so that we can slowly work and enable them.
Suite runs `sqlite` and `mysql` tests. `pgsql` suite needs work (tables are not created? not sure for now)

Thanks

### CODE REVIEW

File: CHANGELOG-4.0.md [hunk 1]
```diff
@@ -1,5 +1,6 @@
 # [4.0.4](https://github.com/phalcon/cphalcon/releases/tag/v4.0.4) (2020-xx-xx)
 ## Added
+- Added a way to utilize GitHub actions to run database tests against each RDBMS and reworked the testing suite. [#14779](https://github.com/phalcon/cphalcon/issues/14779)
```
Comment (ruudboon) [inline CHANGELOG-4.0.md]: I didn't add any comments about our Codeception 4 upgrade. Maybe we should add that here as well.

File: tests/README.md [hunk 2]
```diff
@@ -10,9 +10,12 @@ cp tests/ci/.env.default .env
 cd tests 
 docker-compose up -d
 cd ..
+php tests/_ci/generage-db-schemas.php
```
Comment (ruudboon) [inline tests/README.md]: Can you codecept build here as well.
And maybe we should prefix it with vendor/bin/codecept

File: tests/_config/bootstrap.php [hunk 3]
```diff
@@ -18,7 +18,11 @@
  */
 $root = dirname(dirname(__DIR__)) . DIRECTORY_SEPARATOR;
 
-loadEnvironment($root);
+loadIni();
+loadAutoloader($root);
+loadFolders();
+loadDefined();
+//loadEnvironment($root);
```
Comment (ruudboon) [inline tests/_config/bootstrap.php]: Do we still need this in comments. We can clean it up if you ask me. We can always look back in git. Goes for this whole file

File: tests/README.md [hunk 4]
```diff
@@ -140,15 +143,10 @@ After the compilation is completed, you can check if the extension is loaded:
 ```
```
Comment (ruudboon) [inline tests/README.md]: Can we add some documentation how the new migration models work and how you can add new tables etc.

**Review by ruudboon — CHANGES_REQUESTED** [review]

**Comment by ruudboon** [comment]
Few lines of feedback. But in general great work @niden

**Comment by niden** [comment]
@ruudboon I will check those tonight maybe tomorrow. I have a bit more refactoring to do with the services of the DiTrait and this would be done

**Review by ruudboon — APPROVED** [review]

**Pull request merged**